docs/compatibility.md

```diff
@@ -0,0 +1,183 @@
+# StableHLO Compatibility
+
+## Compatibility Guarantees
+
+(G1) libStablehlo provides 6 months of backward compatibility, i.e. StableHLO programs serialized by an old version of libStablehlo have the same semantics when deserialized by new versions of libStablehlo within the compatibility window.
+
+(G2) libStablehlo provides 3 weeks of forward compatibility, i.e. StableHLO programs serialized by a new version of libStablehlo have the same semantics when deserialized by old versions of libStablehlo within the compatibility window, unless they are using new features introduced since the old version.
+
+(G3) Source compatibility for C, C++ and Python APIs within libStablehlo is an aspirational goal. At the moment, we don't offer source compatibility guarantees, but please let us know if this is an important use case for you, and we can have a discussion about supporting it.
+
+\*StableHLO semantics is defined by the StableHLO specification and can be tested using the StableHLO interpreter. While we're bootstrapping the StableHLO specification, we will be referring to a combination of [XLA's Operation Semantics](https://www.tensorflow.org/xla/operation_semantics) and existing implementations to define the semantics.
+
+## StableHLO Programs
+
+For the purposes of compatibility guarantees, we define StableHLO programs as MLIR programs that include ops, attributes, and types specified below. This definition is based on our experience with lowering JAX/PyTorch/TensorFlow programs to MHLO and interviewing various groups using MHLO. Please let us know if you're interested in expanding this definition, and we can have a discussion about this.
+
+|  Dialect     | Ops, attributes and types |
+| ------------ | ----------- |
+| Arith        | `AddIOp, CmpIOp, CmpIPredicateAttr, ConstantOp, DivSIOp, ExtSIOp, IndexCastOp, MaxSIOp, MinSIOp, MulIOp, SelectOp, SubIOp, TruncIOp` |
+| Builtin      | No ops, but all attributes and types. |
+| CHLO         | All ops, attributes and types. |
+| Func         | `CallOp` |
+| MLProgram    | All ops, attributes and types. |
+| SparseTensor | Aspirational (pending the sparsity RFC which is expected in Q4 2022). |
+| StableHLO    | All ops, attributes and types except `CustomCallOp` whose semantics is implementation-defined. |
+| Tensor       | `CastOp, DimOp, FromElementsOp` |
+
+We will provide a pass `--stablehlo-compatibility-check` which succeeds if a given program is fully covered by these compatibility guarantees and fails if it doesn't.
```

Comment:
recommend calling the pass `verify-stablehlo-compliance` or something with a verb at the beginning.

Reply:
cc: @burmako

I like this name, we originally avoided the word "verify" to avoid overloading that verb. Regardless I agree that starting with a verb is best, either `verify-stablehlo-compliance` or `check-stablehlo-compliance`.

Is using `verify-*` a common convention for this sort of thing?

Reply:
This is simplified in RFC v2. Valid programs consist of only StableHLO ops/types/attributes.